paper2remarkable/providers/epub.py

```diff
@@ -0,0 +1,65 @@
+from ._base import Provider
+from ..utils import chdir, upload_to_remarkable
+import os, tempfile, shutil
+
+class EPUBProvider(Provider):
+    """Provider for direct EPUB uploads"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Override operations since we don't need PDF processing
+        self.operations = []  # No operations needed for direct EPUB upload
+
+    @staticmethod
+    def validate(src):
+        """Validate if source is an EPUB file"""
+        # Convert to absolute path before validation
+        abs_path = os.path.abspath(os.path.expanduser(src))
+        return abs_path.lower().endswith('.epub') and os.path.exists(abs_path)
+
+
+    def get_abs_pdf_urls(self, src):
+        """For EPUB files, just return the local path as absolute path"""
+        abs_path = os.path.abspath(os.path.expanduser(src))
+        return abs_path, abs_path
+
+
+    def run(self, src, filename=None):
```

Comment:
This method duplicates a lot of the logic of the `run`  method of the base class. I understand why this was done (the base class assumes the pdf extension), but I'd like to avoid duplicating the logic if we can. 

This provider is basically the same as the `LocalFile` provider, with the difference that you're (correctly) disabling the pdf operations and setting the `.epub` extension. So I think a simpler approach might be to extract the extension from the source and let the `LocalFile` provider handle things. Here's how that could work:

- In the `Provider` base class, you can extract the file extension from the `clean_filename` variable, and raise an exception if the file extension isn't `pdf`, `ps`, or `epub` (postscript is supported as well)
- Change `self.operations` to a dictionary of operations by extension, with `epub` getting the empty list (as above), and `pdf` and `ps` getting the existing operations list. Then in the `run` method of `Provider` access that list using the identified extension

I think that may be all that's needed for the `LocalFile` provider to support epub as well.